Site/auto deploy

clairesunstudio opened the pull request:
Added:
  - project: Site
    component: CircleCI
    description: Add CI for site for build and deploy. (#1001)
    issue: DP-17980
    impact: Patch

---

Added documentation for branch name prefix
| Branch name prefix  | Rule | Scenario |
|---|---|---|
| `react/`  | This will bypass Patternlab build, deploy and tests | This prefix should only be used if all your changes are in the [react](/react) folder, and nothing in the shared [assets](/assets) folder that could potentially impact the build of the Patternlab implementation  |
| `pattern/`  | This will bypass React Storybook build and tests | This prefix should only be used if all your changes are in the [patternlab](/patternlab) folder, and nothing in the shared [assets](/assets) folder that could potentially impact the build of the React implementation  |
| `site/`  | This will bypass all React and Patternlab and assets build, deploy and tests. |  Only use this prefix if all the changes are in the [site](/site) folder. e.g. Content changes to the homepage site that won't impact React, Patternlab and shared assets. |
| `docs`  | This will bypass all React and Patternlab and assets build, deploy and tests, as well as Mayflower site build and tests. | Only use this prefix if all the changes are in the [docs](/docs) folder. e.g. Documentation changes for the repo and the Mayflower Gitbook that won't impact React, Patternlab, shared assets and the homepage static site. |

clairesunstudio committed:
add changelog

clairesunstudio committed:
Merge branch 'site/auto-deploy' of https://github.com/massgov/mayflower into site/auto-deploy

clairesunstudio committed:
update changelog

clairesunstudio committed:
Merge branch 'develop' into site/auto-deploy

clairesunstudio committed:
Merge branch 'develop' into site/auto-deploy

clairesunstudio committed:
add circle config for site build and deploy

clairesunstudio committed:
fix yarn

clairesunstudio committed:
update circle config

clairesunstudio commented:
Not sure why Github is still showing `patternlab_test` and `react_test` being required. filters are added to the jobs in 58b9478

clairesunstudio committed:
disable assets_test on site branches

clairesunstudio committed:
Merge branch 'develop' into site/auto-deploy

l0c0luke committed:
force ci pipeline

l0c0luke commented on .circleci/config.yml:
```diff
@@ -395,3 +452,8 @@ workflows:
             branches: { ignore: /.*/ }
             tags: { only: /.*/ }
+      - site_deploy_tag:
+          requires: [site_build]
```
Need to add `site_build` job to the tag push workflow.
```
#!/bin/sh -eo pipefail
# site_deploy_tag: requires job \"site_build\" but \"site_build\" is not part of this workflow.
# The following jobs are unreachable: site_deploy_tag
```
from https://app.circleci.com/pipelines/github/massgov/mayflower/52/workflows/fabda5d4-b9d7-461e-88ae-bf63a1f0d17a/jobs/25838

clairesunstudio replied:
good catch!

clairesunstudio committed:
Merge branch 'develop' into site/auto-deploy

clairesunstudio committed:
add site_build to build_tag workflow

l0c0luke committed:
Merge branch 'develop' into site/auto-deploy

l0c0luke commented:
@clairesunstudio `patternlab_test` and `react_test` are mandatory jobs as defined in this repo @rbayliss showed me. It's not a bad idea to run these jobs all the time anyways, just to be sure, so you should remove the ignoring of them.

clairesunstudio committed:
remove branch filters for tests

clairesunstudio committed:
add back filters

clairesunstudio committed:
fix changelog

clairesunstudio committed:
documentation wip

clairesunstudio committed:
add documentation

clairesunstudio commented:
@lukewertz I removed the restrictions from github settings to allow bypassing the tests for none code changes, and added documentation in README.md to specify the use the branch prefixes. Hopefully it makes more sense now.
